### PR TITLE
BN-14 feat: adding item

### PR DESCRIPTION
**Pull request by Charlesntwari**
Trello link:https://trello.com/c/OKJoeWVY/14-a-seller-should-be-able-to-create-add-a-product
# PR Description
As a Seller should be able to add/create items for sale Such that the stock can be seen by any requesting client
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] The item should be added to the sellers collection
- [x] The item should not be duplicated
- [x] Validations
- [x] An appropriate response message should be sent to the requesting client upon success/failure of the transaction
# Description of tasks that were expected to be completed and that are not yet completed
- [x] None
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
3. log in as seller
4. create product
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

File: src/controllers/userController.js [hunk 1]
```diff
@@ -80,20 +73,17 @@ export async function createUser(req, res) {
...
-
   const emailExists = await user.findOne({
     where: {
       email,
     },
   });
-
   if (emailExists) {
     return res.status(409).json({
       status: 400,
```
Comment (Illustre13) [inline src/controllers/userController.js]: I think the status code to be returned here is 409, not 400

File: src/controllers/product/productController.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable camelcase */
```
Comment (Illustre13) [inline src/controllers/product/productController.js]: You should't disable this eslint

**Review by Illustre13 — CHANGES_REQUESTED** [review]

**Review by Niyobuzima — DISMISSED** [review]
**Great Job **

**Review by Honore123 — DISMISSED** [review]
🚀

**Review by Illustre13 — DISMISSED** [review]
Good job Bro!!!

**Review by Honore123 — DISMISSED** [review]
🚀

**Review by Illustre13 — DISMISSED** [review]
Well Done @Charlesntwari !!!

**Review by Niyobuzima — DISMISSED** [review]
**Well Done**

File: src/controllers/userconroller.js [hunk 3]
```diff
@@ -75,12 +75,23 @@ export const updateStatus = catchAsync(async (req, res) => {
...
+        subject: `Your status has been updated to ${status}`,
+        text: `Dear ${user.name}, due to violation of our policies, we regret to inform you that your account has been disactivated.`,
+      });
+    }
+    if (user.status === 'active') {
+      await transporter.sendMail({
+        from: process.env.EMAIL_ADDRESS,
+        to: user.email,
+        subject: `Your status has been updated to ${status}`,
+        text: `Hello ${user.name} hope this email finds you well, our team is pleased to inform you that your accout has been reactivated, your can now access all services we provide to you.`,
```
Comment (dushimeemma) [inline src/controllers/userconroller.js]: Would mind to use a helper function for the email?

File: src/tests/images/Blue and Red Modern Real Estate Instagram Post (1).png
Comment (dushimeemma) [inline src/tests/images/Blue and Red Modern Real Estate Instagram Post (1).png:1]: Clarify the reason of this hardcoded image file

Comment (Charlesntwari) [reply src/tests/images/Blue and Red Modern Real Estate Instagram Post (1).png:1]: i actually followed the assistance you gave me on creating an item with a single image using sinon and then i modified to to receive at least 4 images in order to fit with the requested validation.

File: src/controllers/userconroller.js (hunk 3 above)
Comment (Charlesntwari) [reply src/controllers/userconroller.js]: yeah sure

**Review by Niyobuzima — DISMISSED** [review]

**Review by dushimeemma — COMMENTED** [review]
@Charlesntwari kindly update `.env.example`  and include `the new env variables` you added.

**Review by Honore123 — DISMISSED** [review]

**Review by Niyobuzima — DISMISSED** [review]

**Review by Illustre13 — APPROVED** [review]
Good Job Bro!!!

**Review by Niyobuzima — APPROVED** [review]

**Review by dushimeemma — APPROVED** [review]
Good job 🥇 🥇

**Pull request merged**